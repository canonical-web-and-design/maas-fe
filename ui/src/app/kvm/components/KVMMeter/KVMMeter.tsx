import React from "react";

import Meter from "app/base/components/Meter";

type Props = {
  allocated: number;
  free: number;
  segmented?: boolean;
  total: number;
  unit?: string;
};

const KVMMeter = ({
  allocated,
  free,
  segmented = false,
  total,
  unit = "",
}: Props): JSX.Element | null => {
  return (
    <div className="kvm-meter">
      <div>
        <p className="p-heading--small u-text--light">Total</p>
        <div data-test="total">{`${total}${unit}`}</div>
      </div>
      <div className="u-align--right">
        <p className="p-heading--small u-text--light">
          Allocated
          <span className="u-nudge-left--small">
            <i className="p-circle--link u-no-margin--top"></i>
          </span>
        </p>
        <div
          className="u-nudge-left"
          data-test="allocated"
        >{`${allocated}${unit}`}</div>
      </div>
      <div className="u-align--right">
        <p className="p-heading--small u-text--light">
          Free
          <span className="u-nudge-left--small">
            <i className="p-circle--link-faded u-no-margin--top"></i>
          </span>
        </p>
        <div className="u-nudge-left" data-test="free">{`${free}${unit}`}</div>
      </div>
      <div style={{ gridArea: "meter" }}>
        <Meter
          data={[
            {
              value: allocated,
            },
          ]}
          max={total}
          segmented={segmented}
          small
        />
      </div>
    </div>
  );
};

export default KVMMeter;
