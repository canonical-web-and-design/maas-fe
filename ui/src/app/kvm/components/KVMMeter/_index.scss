@mixin KVMMeter {
  .kvm-meter {
    display: grid;
    grid-column-gap: $sph-inner;
    grid-template-areas:
      "total allocated free"
      "meter meter meter";
    grid-template-columns: repeat(3, minmax(1fr, max-content));
    grid-template-rows: min-content;
  }
}
