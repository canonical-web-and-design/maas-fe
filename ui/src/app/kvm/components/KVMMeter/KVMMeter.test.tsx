import { shallow } from "enzyme";
import React from "react";

import KVMMeter from "./KVMMeter";

describe("KVMMeter", () => {
  it("renders", () => {
    const wrapper = shallow(
      <KVMMeter allocated={2} free={1} total={3} unit="GB" />
    );

    expect(wrapper).toMatchSnapshot();
  });
});
