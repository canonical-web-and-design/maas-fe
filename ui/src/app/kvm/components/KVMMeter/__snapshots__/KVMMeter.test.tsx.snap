// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`KVMMeter renders 1`] = `
<div
  className="kvm-meter"
>
  <div>
    <p
      className="p-heading--small u-text--light"
    >
      Total
    </p>
    <div
      data-test="total"
    >
      3GB
    </div>
  </div>
  <div
    className="u-align--right"
  >
    <p
      className="p-heading--small u-text--light"
    >
      Allocated
      <span
        className="u-nudge-left--small"
      >
        <i
          className="p-circle--link u-no-margin--top"
        />
      </span>
    </p>
    <div
      className="u-nudge-left"
      data-test="allocated"
    >
      2GB
    </div>
  </div>
  <div
    className="u-align--right"
  >
    <p
      className="p-heading--small u-text--light"
    >
      Free
      <span
        className="u-nudge-left--small"
      >
        <i
          className="p-circle--link-faded u-no-margin--top"
        />
      </span>
    </p>
    <div
      className="u-nudge-left"
      data-test="free"
    >
      1GB
    </div>
  </div>
  <div
    style={
      Object {
        "gridArea": "meter",
      }
    }
  >
    <Meter
      data={
        Array [
          Object {
            "value": 2,
          },
        ]
      }
      max={3}
      segmented={false}
      small={true}
    />
  </div>
</div>
`;
